### ISSUE TITLE
Add Logit function

### ISSUE THREAD

**Issue by n-gao** [pytorch/pytorch#37060]
## 🚀 Feature
Add a numerical stable implementation of the logit function, the inverse of the sigmoid function, and its derivative.
```
y = ln(x/(1-x))
```

## Motivation
It should be as easy to use the inverse of the sigmoid as it is to use the sigmoid function without having to worry about a numerical stable implementation.

## Pitch
Add `torch.logit`.

## Alternatives
A current version of the logit could look like this:
```
y = torch.log(x) - torch.log1p(-x)
```

## Additional context
https://de.wikipedia.org/wiki/Logit

**Comment by mruberry** [comment]
Hey @n-gao, thanks for the suggestion! A PR implementing a numerically stable logit would be great!

**Comment by n-gao** [comment]
I'd like to help but am not familiar with the project structure, e.g., I couldn't find the sigmoid implementation.

**Comment by mruberry** [comment]
Hm... maybe we should make a guide for how to add a function to PyTorch.

**Comment by n-gao** [comment]
That would help a lot!

**Comment by riyakothari** [comment]
Is this function being added to pytorch?

**Comment by mruberry** [comment]
It'd be great to have, but I don't think anyone is currently working on it, @riyakothari.

**Comment by BIT-silence** [comment]
I think I can help work on this if no one already started to working on it.

**Comment by BIT-silence** [comment]
I prepared https://github.com/pytorch/pytorch/pull/41062 for this issue. I tested two implements for logit, one is log(x / (1-x)) and another one is log(x) - log1p(-x). Actually in my test the first one has better numerical stability. In the second one, the minus operation may suffer from the [catastrophic cancellation](https://en.wikipedia.org/wiki/Loss_of_significance) when x is around 0.5. Also the first impl is about 20% faster.
